### ISSUE TITLE
Frequent stall and resume of IO observed after Upgrading Pravega Cluster for major versions.

### ISSUE THREAD

**Issue by Richa-Srivastava** [pravega/pravega#4345]
**Once Pravega was upgraded from 0.5.0-2302.7592022 to 0.6.0-2408.7312399, the upgrade was successful and IO was stalled for 2 minutes then resumed back but again after some time IO stalled for around 4 minutes and resumed. This behavior was very frequent and repetitive:**
First time it was stalled for 2 minutes:
```
INFO  [2019-11-07 09:37:47,830] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:19,166, events/sec:25, KB/sec:19.588867
INFO  [2019-11-07 09:37:47,830] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:19,166, events/sec:25, KB/sec:19.588867
...
...
INFO  [2019-11-07 09:40:17,831] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:19,232, events/sec:21, KB/sec:16.38086
INFO  [2019-11-07 09:40:17,832] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:19,298, events/sec:21, KB/sec:16.436523
```
Then again after 23 minutes, it was stalled:
```
INFO  [2019-11-07 10:03:17,831] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:39,237, events/sec:17, KB/sec:13.191406
INFO  [2019-11-07 10:03:17,831] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:59,649, events/sec:26, KB/sec:20.05371
INFO  [2019-11-07 10:03:17,831] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:39,237, events/sec:17, KB/sec:13.191406
INFO  [2019-11-07 10:03:17,831] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:59,649, events/sec:26, KB/sec:20.05371
INFO  [2019-11-07 10:03:47,829] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:39,237, events/sec:16, KB/sec:13.019531
INFO  [2019-11-07 10:03:47,829] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:59,649, events/sec:25, KB/sec:19.792969
...
...
INFO  [2019-11-07 10:06:47,833] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:39,237, events/sec:15, KB/sec:12.079102
INFO  [2019-11-07 10:06:47,833] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:59,649, events/sec:24, KB/sec:18.585938
INFO  [2019-11-07 10:07:17,831] io.pravega.longevity.utils.PerformanceUtils: Writers (1/1): events:39,250, events/sec:15, KB/sec:11.938477
INFO  [2019-11-07 10:07:17,831] io.pravega.longevity.utils.PerformanceUtils: Readers (1/1): events:59,673, events/sec:23, KB/sec:18.151367
```
Following is Pravega Config:
```
bookkeeper:0.6.0-2408.7312399
pravega:0.6.0-2408.7312399
pravega/pravega-operator:0.4.2
pravega/zookeeper:0.2.4
pravega/zookeeper-operator:0.2.4
```
Attaching Metrics also for the same:
![image](https://user-images.githubusercontent.com/48912640/68386695-99724300-0182-11ea-97a4-2c6093ba1ef2.png)

![image](https://user-images.githubusercontent.com/48912640/68386745-b3138a80-0182-11ea-9f07-b7bb3003587b.png)

Attaching k8 logs on the Jira.

**Comment by shrids** [comment]
The test performed is as follows:

Longevity with 0.5.x version of pravega is started.i.e. the Pravega client version used by the longevity worker and Pravega version is 0.5.x

An upgrade of Pravega to 0.6.x is triggered. The longevity worker post the upgrade is still using Pravega client version 0.5.x and is processing responses from Pravega server 0.6.x.

From the logs, we can observe that the longevity worker encounters decoder exceptions post-upgrade. The first such exception is post-upgrade is shown below
```
2019-11-07 09:39:16,382 847153 [epollEventLoopGroup-4-14] WARN  i.p.client.netty.impl.RawClient - Processing failure:
io.pravega.shared.protocol.netty.ConnectionFailedException: io.netty.handler.codec.DecoderException: io.pravega.shared.protocol.netty.InvalidMessageException: Header indicated more bytes than exist.
    at io.pravega.client.netty.impl.FlowHandler.lambda$exceptionCaught$3(FlowHandler.java:222)
    at java.util.concurrent.ConcurrentHashMap.forEach(Unknown Source)
```

The 0.5.x client has the issue https://github.com/pravega/pravega/issues/4211 .  The issue is described below

> If there is an additional field in a WireCommand the protocol is supposed to ignore it to preserve compatibility. Currently, the code ignores the wirecommand, but as the data is un-read netty interprets it as the next command and attempts to parse that, which generally fails, killing the connection.


This issue has been fixed as part of 0.6 (https://github.com/pravega/pravega/pull/4212) but is not present in 0.5.x. Hence we are seeing the `DecoderException` which leads to other incorrect processing causing a chain of other errors on the client-side.

 (worker pod logs)
[workerLogs-debug.zip](https://github.com/pravega/pravega/files/3823896/workerLogs-debug.zip)

**Comment by shrids** [comment]
Closing this as it has been fixed in 0.6 version.
